app/src/main/java/com/eis/phoneringer/structure/RingCommandHandler.java

```diff
@@ -0,0 +1,46 @@
+package com.eis.phoneringer.structure;
+
+import com.eis.smslibrary.SMSPeer;
+
+/**
+ * Class used to check if password in memory corresponds to the password passed through sender
+ *
+ * @author Alberto Ursino, Luca Crema, Marco Mariotto
+ */
+
+public class RingCommandHandler {
+
+    public static final String SPLIT_CHARACTER = "_";
+    private static RingCommandHandler instance = new RingCommandHandler();
+
+    /**
+     * Private constructor
+     */
+    private RingCommandHandler() {
+    }
+
+    /**
+     * @return the singleton
+     */
+    public static RingCommandHandler getInstance() {
+        return instance;
```

Comment:
Singleton doesn't work like this.
You have to create an instance if it's null and then return it, else just return it. So you don't even have to use that `private static RingCommandHandler instance = new RingCommandHandler()`.